osf/models/validators.py

```diff
@@ -38,7 +38,7 @@ def validate_year(item):
         except ValueError:
             raise ValidationValueError('Please enter a valid year.')
         else:
-            if len(item) != 4:
+            if type(item) == str and len(item) != 4:
```

Comment:
Since there's multiple string types in Python (and the names differ between Python 2 and 3), it's best to do this when checking against a string:

```python
from django.utils.six import string_types

isinstance(thing, string_types)
```